src/System.AppContext/src/System/AppContext.cs

```diff
@@ -18,8 +18,7 @@ private enum SwitchValueState
             UnknownValue = 0x8 // Not used on .NET Native
         }
 
-        private static Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
-        private static readonly object s_syncLock = new object();
+        private static readonly Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
```

Comment:
In the Full Framework we use `AppContext` to provide an infrastructure for handling backward compatibility. The basic idea is that each individual library has a set of switches it cares about. Each switch has a default value which it tied to the `TargetFrameworkName` attribute on the assembly.

When a library goes down a path where a decision about the switch is made, the library calls into mscorlib in order to see if the value of the switch has been overriden. If an override value is not set, then the library will fallback to the default values based on the `TargetFrameworkName`.

This allows applications that are not targeting latest framework to selectively opt-in to the new behavior. The multiple states in the enum will handle these various cases.

You are absolutely right that for this specific implementation of AppContext we could have just used a Dictionary<string,bool>. However, when creating this library I wanted to keep the code as much as possible the same.

As to why the default value of the switch defaults to `false`: 
The idea for using AppContext as the way to handle backward compatibility was to have the applications that are targeting the latest framework 'opt-in' to the old behaviors. 

That meant that, by default, on the latest version the value of the flag is `false` but you can opt-in (make it `true`) if you want legacy behavior. 

There was a super tiny performance consideration since the runtime initializes a bool to false by default.